backend/index.js

```diff
@@ -1,54 +1,105 @@
-const express = require('express');
-const cors = require('cors');
-const helmet = require('helmet');
-const http = require('http');
-const responseTime = require('response-time');
-const routes = require('./app/routes');
-const { errorHandler } = require('./helpers/error');
-require('colors');
-require('./helpers/dbConnection');
 require('dotenv').config();
+const cluster = require('cluster');
+const os = require('os');
+
+if (process.env.CLUSTER === 'yes') {
+  const numCPUs = os.cpus().length;
+
+  if (cluster.isMaster) {
+    console.log(`Starting app in cluster mode with ${numCPUs} workers`);
+    console.log(`Master ${process.pid} is running`);
+    for (let i = 0; i < numCPUs; i += 1) {
+      cluster.fork();
+    }
+
+    cluster.on('exit', (worker, code, signal) => {
+      console.log(`worker ${worker.process.pid} died with code: ${code} and signal: ${signal}`);
+      console.log('starting new worker');
+      cluster.fork();
+    });
+
+    return;
+  }
+  // else it's a worker. continue.
+  console.log(`Worker ${process.pid} started`);
+} else {
+  console.log('Starting app in non-cluster mode. (To start in cluster mode, pass CLUSTER=yes in config file)');
+}
+
+const http = require('http');
+const app = require('./app');
+/**
+ * Normalize a port into a number, string, or false.
+ */
+
+function normalizePort(val) {
+  const port = parseInt(val, 10);
+
+  if (Number.isNaN(port)) {
+    // named pipe
+    return val;
+  }
+
```

Comment:
Will go through the rest changes post refactoring

Reply:
Please elaborate the changes you want.

Reply:
Can we not abstract the clustering code to a helper file custer.js?
It will improve readability and help easier onboarding

Reply:
So you want some thing cluster.js to me imported in index.js and used there?

Reply:
Yes, so that the code is well modulated and is quite readable, as of now for a new person he/she would have to go through the entire index.js just to know how the server is starting, but knowing the clustering module is not required to get started hence the abstraction is required

Reply:
@jackfrost13 Ok will try to shift lines 5-27 to a different file. 

Rest of the code is index.js is just to start a http server.

Reply:
@jackfrost13  I have made the necessary changes please look into it.